File: software/zyarm_sdk/python/examples/fast_io_loop.py
import time

from zyarm_sdk import ZyArm, ZyArmConfig


def main() -> None:
    target = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.5]
    with ZyArm(ZyArmConfig(port="/dev/ttyUSB0")) as arm:
        for _ in range(100):
            arm.fast_io(target)
            time.sleep(0.02)


if __name__ == "__main__":
    main()

File: software/zyarm_sdk/cpp/examples/fast_io_loop.cpp
#include <chrono>
#include <thread>

#include "zyarm_sdk/arm.hpp"

int main()
{
  zyarm_sdk::ZyArmConfig config;
  config.port = "/dev/ttyUSB0";
  zyarm_sdk::ZyArm arm(config);
  arm.connect();
  zyarm_sdk::JointArray target{0, 0, 0, 0, 0, 0, 0.5};
  for (int index = 0; index < 100; ++index) {
    arm.fast_io(target);
    std::this_thread::sleep_for(std::chrono::milliseconds(20));
  }
  arm.close();
  return 0;
}
